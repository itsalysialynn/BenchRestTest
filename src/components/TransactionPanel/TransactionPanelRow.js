import React from 'react'
import { object } from 'prop-types'
import { Col } from 'react-flexbox-grid'
import dateFormatHelper from '../../helpers/dateFormatHelper'
import numberConversionHelper from '../../helpers/numberConversionHelper'
import { StyledPanelTableRow, StyledColRight } from '../../styles/StyledPanel'
import { TextMuted, TextRed, TextDark } from '../../styles/StyledText'
import LabelAndData from '../Shared/LabelAndData'

export default function TransactionPanelRow({ transaction }) {
  return (
    <StyledPanelTableRow>
      <Col xs={12} sm={6} md={2} lg={2}>
        <LabelAndData label="Date">
          <TextMuted>{dateFormatHelper.default(transaction.Date)}</TextMuted>
        </LabelAndData>
      </Col>
      <Col xs={12} sm={6} md={4} lg={4}>
        <LabelAndData label="Company">
          <TextDark>{transaction.Company}</TextDark>
        </LabelAndData>
      </Col>
      <Col xs={8} sm={6} md={4} lg={4}>
        <LabelAndData label="Account">
          <TextMuted>{transaction.Ledger}</TextMuted>
        </LabelAndData>
      </Col>
      <StyledColRight className="hidden-sm" xs={4} md={2} lg={2}>
        {transaction.Amount < 0 && (
          <LabelAndData label="Amount">
            <TextRed>
              {numberConversionHelper.toCurrency(transaction.Amount)}
            </TextRed>
          </LabelAndData>
        )}
        {transaction.Amount >= 0 && (
          <LabelAndData label="Amount">
            <TextDark>
              {numberConversionHelper.toCurrency(transaction.Amount)}
            </TextDark>
          </LabelAndData>
        )}
      </StyledColRight>
      <Col className="hidden-xs hidden-md hidden-lg hidden-xl" sm={6}>
        {transaction.Amount < 0 && (
          <LabelAndData label="Amount">
            <TextRed>
              {numberConversionHelper.toCurrency(transaction.Amount)}
            </TextRed>
          </LabelAndData>
        )}
        {transaction.Amount >= 0 && (
          <LabelAndData label="Amount">
            <TextDark>
              {numberConversionHelper.toCurrency(transaction.Amount)}
            </TextDark>
          </LabelAndData>
        )}
      </Col>
    </StyledPanelTableRow>
  )
}

TransactionPanelRow.propTypes = {
  transaction: object.isRequired,
}