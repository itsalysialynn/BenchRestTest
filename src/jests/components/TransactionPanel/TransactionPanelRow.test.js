import React from 'react'
import { shallow } from 'enzyme'
import TransactionPanelRow from '../../../components/TransactionPanel/TransactionPanelRow'

/* global describe, it, expect, test, jest, test  */

const props = {
  transaction: {
    Date: '2016-01-29T17:35:00.000-07:00',
    Amount: '-5',
    Ledger: 'Interest Earned',
    Company: 'TD'
  },
}

describe('rendering transaction panel', () => {
  const component = shallow(<TransactionPanelRow {...props} />)
  expect(component.html()).toMatch('Interest Earned')
  expect(component.html()).toMatch('Jan 29th, 2016')
  expect(component.html()).toMatch('($5.00)')
  expect(component.html()).toMatch('TD')

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})


