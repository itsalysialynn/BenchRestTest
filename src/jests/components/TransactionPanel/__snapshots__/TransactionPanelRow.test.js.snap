// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering transaction panel should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TransactionPanelRow
    transaction={
      Object {
        "Amount": "-5",
        "Company": "TD",
        "Date": "2016-01-29T17:35:00.000-07:00",
        "Ledger": "Interest Earned",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Col
          lg={2}
          md={2}
          sm={3}
          xs={6}
        >
          <Styled(styled.div)>
            Jan 29th, 2016
          </Styled(styled.div)>
        </Col>,
        <Col
          lg={4}
          md={4}
          sm={6}
          xs={6}
        >
          <Styled(styled.div)>
            TD
          </Styled(styled.div)>
        </Col>,
        <Col
          lg={4}
          md={4}
          sm={3}
          xs={6}
        >
          <Styled(styled.div)>
            Interest Earned
          </Styled(styled.div)>
        </Col>,
        <Styled(Col)
          lg={2}
          md={2}
          sm={3}
          xs={6}
        >
          <Styled(styled.div)>
            ($5.00)
          </Styled(styled.div)>
        </Styled(Col)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Styled(styled.div)>
            Jan 29th, 2016
          </Styled(styled.div)>,
          "lg": 2,
          "md": 2,
          "sm": 3,
          "xs": 6,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Jan 29th, 2016",
          },
          "ref": null,
          "rendered": "Jan 29th, 2016",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Styled(styled.div)>
            TD
          </Styled(styled.div)>,
          "lg": 4,
          "md": 4,
          "sm": 6,
          "xs": 6,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "TD",
          },
          "ref": null,
          "rendered": "TD",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Styled(styled.div)>
            Interest Earned
          </Styled(styled.div)>,
          "lg": 4,
          "md": 4,
          "sm": 3,
          "xs": 6,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Interest Earned",
          },
          "ref": null,
          "rendered": "Interest Earned",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(styled.div)>
              ($5.00)
            </Styled(styled.div)>,
            false,
          ],
          "lg": 2,
          "md": 2,
          "sm": 3,
          "xs": 6,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "($5.00)",
            },
            "ref": null,
            "rendered": "($5.00)",
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Col
            lg={2}
            md={2}
            sm={3}
            xs={6}
          >
            <Styled(styled.div)>
              Jan 29th, 2016
            </Styled(styled.div)>
          </Col>,
          <Col
            lg={4}
            md={4}
            sm={6}
            xs={6}
          >
            <Styled(styled.div)>
              TD
            </Styled(styled.div)>
          </Col>,
          <Col
            lg={4}
            md={4}
            sm={3}
            xs={6}
          >
            <Styled(styled.div)>
              Interest Earned
            </Styled(styled.div)>
          </Col>,
          <Styled(Col)
            lg={2}
            md={2}
            sm={3}
            xs={6}
          >
            <Styled(styled.div)>
              ($5.00)
            </Styled(styled.div)>
          </Styled(Col)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Styled(styled.div)>
              Jan 29th, 2016
            </Styled(styled.div)>,
            "lg": 2,
            "md": 2,
            "sm": 3,
            "xs": 6,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Jan 29th, 2016",
            },
            "ref": null,
            "rendered": "Jan 29th, 2016",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Styled(styled.div)>
              TD
            </Styled(styled.div)>,
            "lg": 4,
            "md": 4,
            "sm": 6,
            "xs": 6,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "TD",
            },
            "ref": null,
            "rendered": "TD",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Styled(styled.div)>
              Interest Earned
            </Styled(styled.div)>,
            "lg": 4,
            "md": 4,
            "sm": 3,
            "xs": 6,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Interest Earned",
            },
            "ref": null,
            "rendered": "Interest Earned",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(styled.div)>
                ($5.00)
              </Styled(styled.div)>,
              false,
            ],
            "lg": 2,
            "md": 2,
            "sm": 3,
            "xs": 6,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "($5.00)",
              },
              "ref": null,
              "rendered": "($5.00)",
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
